File: Java/Project Euler/004/fourth.java
class fourth{
	public static void main(String[]argv){
		prog p1 = new prog();
		p1.runCode();
	}
}

class prog{
	
	int result = 0;
	String testInt = "";
	boolean found = false;
	int maxPal =-1;
	void runCode(){
		for(int i=999; i>99; i--){
			for(int j = 999; j>99; j--){
				testInt = ""+(i*j);
				if(isPalindrome(testInt) && ((i*j)>maxPal)){
					maxPal= i*j;
				}
			}
		}
		System.out.println("\n" + maxPal);
	}
	
	boolean isPalindrome(String str) {
		return str.equals(new StringBuilder(str).reverse().toString());
	}
}

File: Python/Project Euler/004/fourth.py
def findPalindrome():
	result = 0
	for num in range(999, 99, -1):
		for num2 in range(999, 99, -1):
			if(isPalindrome(str(num*num2)) == 1 and ((num*num2) > result)):
			result = (num*num2)
	print result


def isPalindrome(num):
	if(num == (num[::-1])):
		return 1
	return 0
	
findPalindrome()